services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -850,9 +849,19 @@ const addHeightToIndexBlocksQueue = async (height, priority) => {
 		: indexBlocksQueue.add({ height });
 };
 
-const setIndexVerifiedHeight = ({ height }) => keyValueTable.set(INDEX_VERIFIED_HEIGHT, height);
+const getIndexVerifiedHeight = async () => {
+	const blocksTable = await getBlocksTable();
+	const [lastIndexedFinalBlock = {}] = await blocksTable.find(
+		{
+			isFinal: true,
+			sort: 'height:desc',
+			limit: 1,
+		},
+		['height'],
+	);
 
-const getIndexVerifiedHeight = () => keyValueTable.get(INDEX_VERIFIED_HEIGHT);
+	 return lastIndexedFinalBlock.height || getGenesisHeight();
```

Comment:
Should return null if nothing is indexed.

```suggestion
	 return lastIndexedFinalBlock.height || null;
```

Reply:
Applied.